synapseutils/migrate_functions.py

```diff
@@ -350,16 +350,23 @@ def _ensure_schema(cursor):
     # we get counts grouping on status
     cursor.execute("create index if not exists ix_status on migrations(status)")
 
+    # we check to see if there is already a migrated copy of a file handle before doing a copy
+    cursor.execute(
+        "create index if not exists ix_file_handle_ids on migrations(from_file_handle_id, to_file_handle_id)"
```

Comment:
It isn't 1-1 since we're multiple entity/versions can share a file handle id. It's probably the case that the index could only have the from value since the number of entities sharing a file handle is probably not large, but the code does query on both these values (from_file_handle_id matches and to_file_handle_id is not null in order to find the first row where the file handle is filled in).